worksheets_test.go

```diff
@@ -84,6 +86,227 @@ func (s *Zuite) TestWorksheetNew_multipleDefsSameName() {
+	value, _ := NewValue(strconv.FormatInt(2018-birthYear, 10))
+	return value
+}
+
+type bio []string
+
+var _ ComputedBy = bio([]string{})
+
+func (fn bio) Args() []string {
+	return fn
+}
+
+func (fn bio) Compute(values ...Value) Value {
+	fullName := values[0].String()
+	birthYear := values[1].String()
+	age := values[2].String()
+	fullName = fullName[1 : len(fullName)-1]
+
+	return NewText(fmt.Sprintf("%s, age %s, born in %s", fullName, age, birthYear))
+}
+
+func (s *Zuite) TestExample_externalComputedBy4point5() {
+	opt := Options{
+		Plugins: map[string]map[string]ComputedBy{
+			"simple": map[string]ComputedBy{
+				"name": sayAlice([]string{}),
+			},
+		},
+	}
+	_, err := NewDefinitions(strings.NewReader(`worksheet simple {
+		1:name text computed_by { external }
+		2:age number[0]
+	}`), opt)
+	if assert.Error(s.T(), err) {
+		require.Equal(s.T(), "plugins: simple.name plugin has no dependencies", err.Error())
+	}
+}
+
+func (s *Zuite) TestExample_externalComputedBy5() {
+	opt := Options{
+		Plugins: map[string]map[string]ComputedBy{
+			"simple": map[string]ComputedBy{
+				"name": sayAlice([]string{"agee"}),
+			},
+		},
+	}
+	_, err := NewDefinitions(strings.NewReader(`worksheet simple {
+		1:name text computed_by { external }
+		2:age number[0]
+	}`), opt)
+	if assert.Error(s.T(), err) {
+		require.Equal(s.T(), "plugins: simple.name plugin has incorrect arg agee", err.Error())
+	}
+}
+
+func (s *Zuite) TestExternalComputedBy_good() {
+	opt := Options{
+		Plugins: map[string]map[string]ComputedBy{
+			"simple": map[string]ComputedBy{
+				"name": sayAlice([]string{"age"}),
+			},
+		},
+	}
+	defs, err := NewDefinitions(strings.NewReader(`worksheet simple {
+		1:name text computed_by { external }
+		2:age number[0]
+	}`), opt)
+	require.NoError(s.T(), err)
+
+	ws := defs.MustNewWorksheet("simple")
+
+	require.False(s.T(), ws.MustIsSet("name"))
+
+	ws.MustSet("age", MustNewValue("73"))
+	require.Equal(s.T(), `"Alice"`, ws.MustGet("name").String())
+}
+
+func (s *Zuite) TestExternalComputedBy_goodComplicated() {
+	opt := Options{
+		Plugins: map[string]map[string]ComputedBy{
+			"complicated": map[string]ComputedBy{
+				"full_name": fullName([]string{"first_name", "last_name"}),
+				"age":       age([]string{"birth_year"}),
+				"bio":       bio([]string{"full_name", "birth_year", "age"}),
+			},
+		},
+	}
+	defs, err := NewDefinitions(strings.NewReader(`worksheet complicated {
+		1:first_name text
+		2:last_name text
+		3:full_name text computed_by { external }
+		4:birth_year number[0]
+		5:age number[0] computed_by { external }
+		6:bio text computed_by { external }
+	}`), opt)
+	require.NoError(s.T(), err)
+
+	ws := defs.MustNewWorksheet("complicated")
+
+	require.False(s.T(), ws.MustIsSet("full_name"))
+
+	ws.MustSet("first_name", NewText("Alice"))
+	ws.MustSet("last_name", NewText("Maters"))
+	ws.MustSet("birth_year", MustNewValue("1945"))
+	require.Equal(s.T(), `"Alice Maters, age 73, born in 1945"`, ws.MustGet("bio").String())
```

Comment:
Sweet!

Can you also add assertions on `full_name` and on `age` before the full `bio`?